cromwell-backend/src/main/scala/cromwell/backend/config/BackendConfiguration.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.backend.config
+
+import com.typesafe.config.ConfigFactory
+
+import scala.collection.JavaConversions._
+
+/**
+  * Defines a backend configuration.
+  * @param name Backend name.
+  * @param initClass Initialization class for the specific backend.
+  */
+case class BackendConfigurationEntry(name: String, initClass: String)
+
+object BackendConfiguration {
+  val config = ConfigFactory.load()
+  val backendCfg = config.getConfig("backend")
+  val defaultBackend = backendCfg.getString("default")
+  val backendProviders = backendCfg.getConfigList("providers")
+  val backendList = backendProviders.map(entry =>
+    BackendConfigurationEntry(entry.getString("name"), entry.getString("initClass"))).toList
+
+  def apply(): BackendConfiguration = new BackendConfiguration(backendList, defaultBackend)
+}
+
+/**
+  * Retrieves backend configuration.
+  * @param backendList List of backend entries in configuration file.
+  * @param defaultBackend Backend name to be used as default.
+  */
+class BackendConfiguration(backendList: List[BackendConfigurationEntry], defaultBackend: String) {
```

Comment:
Hey @francares - I was just talking to @kshakir and believe I understand what he's saying (minus my side-philosophical discussion there) and I agree with his point.

There are a few things going on here:

* You're using JavaConversions instead of JavaConverters. The latter is generally preferred as it's using an impliticly enhanced method instead of a direct impliclit conversion
* That'll implicitly convert your java.util.List to a traversable and that's what you're running the combinator over, causing you to need the toList at the end. 

I think the right thing to do here would be to use JavaConverters and then do `val backendProviders = backendCfg.getConfigList("providers").asScala.toList`  - the toList is necessary as you'll end up with a mutable.Buffer I believe from the asScala. Now you have a proper scala List right at the moment of import.